.github/mergify.yml

```diff
@@ -0,0 +1,17 @@
+pull_request_rules:
+  - name: Close on changes only to EOL folders
+    conditions:
+      - and:
+        - label=end-of-life
+        # Active ROS 1 distros
+        - label!=melodic
+        - label!=noetic
+        # Active ROS 2 distros
+        - label!=foxy
+        - label!=galactic
+        - label!=rolling
+        # Other labels
+        - label!=rosdep
+    actions:
+      close:
+        message: This pull request has been automatically closed by Mergify since it only changes files for a ROS distribution that is no longer supported (End Of Life) and the distribution will not have any future syncs to include this change.
```

Comment:
I'm not sure if it's possible, but pointing out in the message what EOL distros are causing the issue would be awesome.

We could also always leave a message when there's a change to a EOL folder, but only close the PR if only EOL distros were modified (I mean, the check also leaves a message in https://github.com/audrow/rosdistro/pull/5, but doesn't close the PR).
I think contributors are much more likely to read a message that to look at the labels.

Reply:
> We could also always leave a message when there's a change to a EOL folder, but only close the PR if only EOL distros were modified (I mean, the check also leaves a message in [audrow#5](https://github.com/audrow/rosdistro/pull/5), but doesn't close the PR).

I've set this up. Thanks for the idea.